manifests/agent/service.pp

```diff
@@ -29,11 +31,23 @@
 
       $times = ip_to_cron($puppet::runinterval)
 
-      cron { 'puppet':
-        command => $command,
-        user    => root,
-        hour    => $times[0],
-        minute  => $times[1],
+      if $::osfamily == 'windows' {
+        scheduled_task { 'puppet':
+          ensure  => present,
+          enabled => true,
+          command => $command,
+          trigger => {
+            schedule   => daily,
+            start_time => $times[0],
```

Comment:
Maybe just error out of the user tries to set cron on windows? Then on absent you can just hide the cron part inside the if block. That way we only support puppet as deamon on windows which IMHO is good enough for now.